src/main/cpp/chassis/headingStates/SpecifiedHeading.cpp

```diff
@@ -26,17 +26,29 @@ SpecifiedHeading::SpecifiedHeading() : ISwerveDriveOrientation(ChassisOptionEnum
 {
 }
 
+SpecifiedHeading::SpecifiedHeading(ChassisOptionEnums::HeadingOption option) : ISwerveDriveOrientation(option),
+                                                                               m_targetAngle(units::angle::degree_t(0.0))
+{
+}
 void SpecifiedHeading::UpdateChassisSpeeds(ChassisMovement &chassisMovement)
 {
-    m_targetAngle = chassisMovement.yawAngle;
-    Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "trajectory drive path planner Pose2d", "Target Angle Specified", m_targetAngle.to<double>());
-
-    chassisMovement.chassisSpeeds.omega += CalcHeadingCorrection(m_targetAngle, m_kPGoalHeadingControl);
+    m_targetAngle = GetTargetAngle(chassisMovement);
 
     auto config = ChassisConfigMgr::GetInstance()->GetCurrentConfig();
     auto chassis = config != nullptr ? config->GetSwerveChassis() : nullptr;
     if (chassis != nullptr)
     {
-        chassis->SetStoredHeading(chassis->GetPose().Rotation().Degrees());
+        auto error = chassis->GetPose().Rotation().Degrees() - m_targetAngle;
+        auto slot = abs(error.value()) < m_fineCoarseAngle.value() ? m_fineSlot : m_coarseSlot;
+        auto correction = CalcHeadingCorrection(m_targetAngle, kPSpecifiedHeading[slot]);
+        chassisMovement.chassisSpeeds.omega += correction;
+        // Not sure if this should be done as its inconsistent with maintain, but face speaker had it
+        // I'm inclined to believe this is correct as we now have a desired heading we want to maintain
+        chassis->SetStoredHeading(m_targetAngle);
```

Comment:
Yeah we need to store the heading or if you leave specified angel you will go to the last stored heading and whip around. The only thing I think we should do/clean up is remove specified heading/Angel and just use maintain. But that might be a off season thing. Think it is fine the way it is

Reply:
deleted comment